src/NServiceBus.Core/Pipeline/PipelineSettings.cs

```diff
@@ -76,11 +66,62 @@ public void Replace<T>(string stepId, Func<IServiceProvider, T> factoryMethod, s
             modifications.Replacements.Add(new ReplaceStep(stepId, typeof(T), description, b => factoryMethod(b)));
         }
 
+        /// <summary>
+        /// Registers a new step behavior or replaces the existing one with the same id with a new one.
+        /// </summary>
+        /// <param name="stepId">The identifier of the step to replace its implementation.</param>
+        /// <param name="behavior">The new <see cref="Behavior{TContext}" /> to use.</param>
+        /// <param name="description">The description of the new behavior.</param>
+        public void RegisterOrReplace(string stepId, Type behavior, string description = null)
+        {
+            BehaviorTypeChecker.ThrowIfInvalid(behavior, nameof(behavior));
+            Guard.AgainstNullAndEmpty(nameof(stepId), stepId);
+            EnsureWriteEnabled(stepId, nameof(Replace));
+            EnsureWriteEnabled(stepId, nameof(Register));
+
+            modifications.AdditionsOrReplacements.Add(AddOrReplaceStep.Create(stepId, behavior, description));
+        }
+
+        /// <summary>
+        /// Registers a new step behavior or replaces the existing one with the same id with a new one.
+        /// </summary>
+        /// <param name="stepId">The identifier of the step to replace its implementation.</param>
+        /// <param name="behavior">The new <see cref="Behavior{TContext}" /> to use.</param>
+        /// <param name="description">The description of the new behavior.</param>
+        public void RegisterOrReplace<T>(string stepId, T behavior, string description = null)
+            where T : IBehavior
+        {
+            BehaviorTypeChecker.ThrowIfInvalid(typeof(T), nameof(behavior));
+            Guard.AgainstNullAndEmpty(nameof(stepId), stepId);
+            EnsureWriteEnabled(stepId, nameof(Replace));
+            EnsureWriteEnabled(stepId, nameof(Register));
+
+            modifications.AdditionsOrReplacements.Add(AddOrReplaceStep.Create(stepId, typeof(T), description,builder => behavior));
+        }
+
+        /// <summary>
+        /// Registers a new step behavior or replaces the existing one with the same id with a new one.
+        /// </summary>
+        /// <param name="stepId">The identifier of the step to replace its implementation.</param>
+        /// <param name="factoryMethod">The factory method to create new instances of the behavior.</param>
+        /// <param name="description">The description of the new behavior.</param>
+        public void RegisterOrReplace<T>(string stepId, Func<IServiceProvider, T> factoryMethod, string description = null)
+            where T : IBehavior
+        {
+            BehaviorTypeChecker.ThrowIfInvalid(typeof(T), "behavior");
```

Comment:
Shouldn't that be `nameof(T)`? Not sure if that works.

Reply:
The `ThrowIfInvalid`-method expects a Type parameter, so `nameof(T)` wouldn't work as it would return the type name